probe-rs/src/architecture/arm/core/cortex_m.rs

```diff
@@ -131,6 +131,42 @@ impl MemoryMappedRegister for Cpacr {
     const NAME: &'static str = "CPACR";
 }
 
+bitfield! {
+    #[derive(Copy, Clone)]
+    /// IPSR - Interrupt Program Status Register (a variant of xPSR)
+    pub struct Ipsr(u32);
```

Comment:
We already have the `Xpsr` register, which also contains this part of the register. Would it be possible to just use that?

Reply:
Fair comment. When I started with this, I didn't know how much Ipsr specific code there would be, so I decided to 'localize' the initial code impact while I experiment. I will keep your question in mind (please don't mark this as resolved) and when I am in code cleanup phase, I will see if it makes sense to move this code to the Xpsr implementation.